sharding-orchestration/sharding-orchestration-core-new/src/main/java/org/apache/shardingsphere/orchestration/temp/internal/registry/ShardingOrchestrationFacade.java

```diff
@@ -0,0 +1,130 @@
+        stateService = new StateService(registryCenterConfiguration.getNamespace(), regCenter);
+        InstanceConfiguration configCenterConfiguration = getConfigurationByOrchestrationType(orchestrationConfig.getInstanceConfigurationMap(),
+                "config_center");
+        configCenter = new ConfigCenterServiceLoader().load(configCenterConfiguration);
+        configService = new ConfigurationService(configCenterConfiguration.getNamespace(), configCenter);
+        listenerManager = shardingSchemaNames.isEmpty() 
+                ? new ShardingOrchestrationListenerManager(registryCenterConfiguration.getNamespace(), regCenter,
+                        configCenterConfiguration.getNamespace(), configCenter, configService.getAllShardingSchemaNames())
+                : new ShardingOrchestrationListenerManager(registryCenterConfiguration.getNamespace(), regCenter, 
+                configCenterConfiguration.getNamespace(), configCenter, shardingSchemaNames);        
+    }
+    
+    /**
+     * Initialize for orchestration.
+     *
+     * @param dataSourceConfigurationMap schema data source configuration map
+     * @param schemaRuleMap schema rule map
+     * @param authentication authentication
+     * @param props properties
+     */
+    public void init(final Map<String, Map<String, DataSourceConfiguration>> dataSourceConfigurationMap,
+                     final Map<String, RuleConfiguration> schemaRuleMap, final Authentication authentication, final Properties props) {
+        for (Entry<String, Map<String, DataSourceConfiguration>> entry : dataSourceConfigurationMap.entrySet()) {
+            configService.persistConfiguration(entry.getKey(), dataSourceConfigurationMap.get(entry.getKey()), schemaRuleMap.get(entry.getKey()), authentication, props, isOverwrite);
+        }
+        stateService.persistInstanceOnline();
+        stateService.persistDataSourcesNode();
+        listenerManager.initListeners();
+    }
+    
+    /**
+     * Initialize for orchestration.
+     */
+    public void init() {
+        stateService.persistInstanceOnline();
+        stateService.persistDataSourcesNode();
+        listenerManager.initListeners();
+    }
+    
+    @Override
+    public void close() {
+        try {
+            regCenter.close();
+            // CHECKSTYLE:OFF
+        } catch (final Exception ex) {
+            // CHECKSTYLE:ON
+            log.warn("RegCenter exception for: {}", ex.getMessage());
+        }
+    }
+    
+    private InstanceConfiguration getConfigurationByOrchestrationType(final Map<String, InstanceConfiguration> map, final String type) {
+        if (null == map || null == type) {
+            return null;
+        }
+        for (Entry entry: map.entrySet()) {
+            InstanceConfiguration configuration = (InstanceConfiguration) entry.getValue();
+            if (type.equals(configuration.getOrchestrationType())) {
+                return configuration;
```

Comment:
`Entry<String, InstanceConfiguration>`  and don't need Class cast.

Reply:
THX. I've got it.